### PR TITLE
Fix issues in Python3/PyQt5.5

### LINKED ISSUE #2573: Spyder is crashing with PyQt5.5.0 on Windows / Python 3.4
Hi dear Spyder team,

When I try to start Spyder 3.0.0dev of this morning :
- first time is ok,
- second time I get a crash, which requires a spyder --reset.

Any idea if it's me/Pyqt5/spyder problem ?

````
Traceback (most recent call last):
  File "D:\WinPythonQt5\basedir34\build\winpython-3.4.3.5_build6Qt5.amd64\python-3.4.3.amd64\lib\site-packages\spyderlib\spyder.py", line 3180, in main
    mainwindow = run_spyder(app, options, args)
  File "D:\WinPythonQt5\basedir34\build\winpython-3.4.3.5_build6Qt5.amd64\python-3.4.3.amd64\lib\site-packages\spyderlib\spyder.py", line 3066, in run_spyder
    main.setup()
  File "D:\WinPythonQt5\basedir34\build\winpython-3.4.3.5_build6Qt5.amd64\python-3.4.3.amd64\lib\site-packages\spyderlib\spyder.py", line 846, in setup
    self.editor = Editor(self)
  File "D:\WinPythonQt5\basedir34\build\winpython-3.4.3.5_build6Qt5.amd64\python-3.4.3.amd64\lib\site-packages\spyderlib\plugins\editor.py", line 462, in __init__
    self.splitter.restoreState( QByteArray().fromHex(str(state)) )
TypeError: QByteArray.fromHex(QByteArray): argument 1 has unexpected type 'str'
````
A wild guess seems maybe when I close spyder something doesn't close/cleanup: the "restart" option fails too

### PR DESCRIPTION
**Pull request by cy18**
Fixes #2573 

----

Fix issue that codes like QByteArray().fromHex(str(state)) raise Type Error in Python3/PyQt5.5

A Type Error is raised when codes like QByteArray().fromHex(str(state)) are called.

It seems the error is because in python3, str(state) will return an string while the QByteArray().fromHex expect an hex array. 
I search the whole project of spyder and modifed codes like  QByteArray().fromHex(str(state)) to QByteArray().fromHex(str(state).encode('utf-8')). The encode method of string will return a byte array in python3 and a string in python2. 

I'm not familiar with  PyQt and the source code of spyder. Feel free to reject my pull request if you have a better solution for this issue.

### CODE REVIEW

**Comment by goanpeca** [comment]
@cy18 thanks for working on this :-) 

@stonebig can you check with this PR if your problem is solved?

File: spyderlib/plugins/editor.py [hunk 1]
```diff
@@ -93,14 +93,14 @@ def clear_breakpoint(filename, lineno):
 class EditorConfigPage(PluginConfigPage):
     def get_name(self):
         return _("Editor")
-
```
Comment (ccordoba12) [inline spyderlib/plugins/editor.py]: Ok, first comment: please revert **all** this blank space changes :-) This makes merging other PRs really, really hard.

**Comment by ccordoba12** [comment]
@stonebig, could you test this PR? It should solve issue #2573.

**Comment by cy18** [comment]
I have reverted the changes about spaces.

File: spyderlib/widgets/sourcecode/base.py [hunk 2]
```diff
@@ -1093,10 +1093,16 @@ def wheelEvent(self, event):
...
+                if hasattr(event, 'angleDelta'):
+                    if event.angleDelta().y() < 0:
+                        self.zoom_out.emit()
+                    elif event.angleDelta().y() > 0:
+                        self.zoom_in.emit()
+                elif hasattr(event, 'delta'):
+                    if event.delta() < 0:
+                        self.zoom_out.emit()
+                    elif event.delta() > 0:
+                        self.zoom_in.emit()
```
Comment (ccordoba12) [inline spyderlib/widgets/sourcecode/base.py]: Why this change? I say it because it's not related to `QByteArray`

Comment (cy18) [inline spyderlib/widgets/sourcecode/base.py]: It's another bug related to PyQt5.5 . I fixed it and created a new pull request. I don't know why it is shown here...

Comment (ccordoba12) [inline spyderlib/widgets/sourcecode/base.py]: Don't worry, let's consolidate all PyQt5.5 issues here

**Comment by ccordoba12** [comment]
Thanks for your work @cy18! I'm going to merge this so we can release our first beta of Spyder 3.0 ;-)

Please open new PRs if you find more issues with PyQt5.5
